rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,727 @@
+            schema:
+              $ref: '#/components/schemas/SetPropertiesRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                example: '{ data: { success: true }, error: { } }'
+  # TODO - Pagination
+  /v1/namespaces:
+    parameters:
+      - name: parent
+        in: query
+        description: Optional parent namespace under which to list namespaces. When empty, list top-level namespaces.
+        required: false
+        schema:
+          type: string
+    get:
+      tags:
+        - Catalog API
+      summary: List namespaces, optionally providing a parent namespace to list underneaath
+      description:
+        List all namespaces at a certain level, optionally starting from a given parent namespace.
+        For example, if table a.b.t exists, using 'SELECT NAMESPACE IN a' this would translate into
+        `GET /v1/namespaces?parent=a` and must return Namepace.of("a","b").
+      operationId: listNamespaces
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListNamespacesResponse'
+        '401':
+          description: Unauthorized
+  /v1/namespaces/{namespace}:
+    parameters:
+      - name: namespace
+        in: path
+        required: true
+        schema:
+          type: string
+    post:
+      tags:
+        - Catalog API
+      summary: Create a namespace
+      description: Create a namespace, with an optional set of properties. The server might also add properties.
+      operationId: createNamespace
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateNamespaceRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+        '401':
+          description: Unauthorized
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a namespace from the catalog. Namespace must be empty.
+      operationId: dropNamespace
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: boolean
+        # TODO - Not empty exception, No such namespace exception
+        '401':
+          description: Unauthorized
+components:
+  schemas:
+    CommitTableRequest:
+      type: object
+      properties:
+        tableIdentifier:
+          $ref: '#/components/schemas/TableIdentifier'
+        metadataJson:
+          type: string
+    TableIdentifier:
+      type: object
+      required:
+        - namespace
+      properties:
+        namespace:
+          type: array
+          items:
+            type: string
+          nullable: false  # Note this is ignored
+        name:
+          pattern: \S
```

Comment:
What's this pattern? No spaces? Should that be some identifierPart type that we use for namespaces as well?

Reply:
I believe that’s just any character, including spaces. I’ll look into what form of regex it is.